rd-net/Lifetimes/Threading/ByteBufferAsyncProcessor.cs

```diff
@@ -463,6 +463,9 @@ public bool Terminate(int timeoutMs = Timeout.Infinite)
     }
 
 
+#if !NET35
+    [System.Runtime.ExceptionServices.HandleProcessCorruptedStateExceptions] // to force myLock to be unlocked even in case of corrupted state exception
```

Comment:
I would have also added logging of the exception

Reply:
Thanks, a good idea! See #190.